subprojects/junit-jupiter/junit-jupiter.gradle

```diff
@@ -0,0 +1,17 @@
+description = "Mockito JUnit 5 support"
+
+apply from: "$rootDir/gradle/java-library.gradle"
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+dependencies {
+    compile project.rootProject
+    testCompile libraries.assertj
+    testCompile libraries.junitJupiterLauncher
+    implementation libraries.junitJupiterEngine
```

Comment:
@TimvdLippe I know it's a bit late now... but depending on junit-jupiter-api should be sufficient, shouldn't it?

Reply:
As types like `org.junit.platform.engine.discovery.DiscoverySelectors` and `org.junit.platform.engine.TestExecutionResult` are used within the tests at least a "test-related" configuration needs the `junit-jupiter-engine` dependency.

Reply:
Um, I am not sure. There are many artifacts and I am not sure what is the minimum we must support. If api is sufficient, feel free to send in a PR :)

Reply:
The current published [pom.xml](https://bintray.com/mockito/maven/download_file?file_path=org%2Fmockito%2Fmockito-junit-jupiter%2F2.17.0%2Fmockito-junit-jupiter-2.17.0.pom) contains the following mandatory transitive dependency.

```xml
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-engine</artifactId>
      <version>5.1.0</version>
      <scope>runtime</scope>
    </dependency>
```

I agree with @marcphilipp and @sormuras: you should likely have a mandatory `compile` dependency on `junit-jupiter-api` and an `optional` `runtime` dependency on `junit-jupiter-engine`.

Reply:
Developers will of course need the dependency on the engine to actually execute tests, but I'm not so sure that the Mockito artifact should assume the developer wants that as a mandatory dependency.

Some users prefer to configure the dependency on the engine within the plug-in configuration for Maven Surefire as opposed to having it as a top-level dependency for the entire project.

Reply:
Any volunteers for creating such a PR?

Reply:
We have already updated the dependency in #1344

Reply:
... or perhaps just a `test` dependency on `junit-jupiter-engine`, since Mockito only depends on it within Mockito's own test suite.

Reply:
Ahhh... I missed #1344.

Thanks for providing the link!